cmds/uinit/uinit.go

```diff
@@ -232,8 +234,34 @@ func dologin() error {
 	return nil
 }
 
+func homedir() error {
+	f, err := os.Open("/usr/user.cpio")
+	if err != nil {
+		return err
+	}
+	archiver, err := cpio.Format("newc")
+	if err != nil {
+		return err
+	}
+	rr := archiver.Reader(f)
+	for {
+		rec, err := rr.ReadRecord()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error reading records: %v", err)
+		}
+		debug("Creating %s\n", rec)
+		if err := cpio.CreateFile(rec); err != nil {
+			log.Printf("Creating %q failed: %v", rec.Name, err)
+		}
+	}
+	return nil
+}
+
 func xrunuser() error {
-	for _, f := range []string{"wingo", "AppChrome", "chrome", "upspin_sos"} {
+	for _, f := range []string{"wingo", "AppChrome", "upspin_sos", "chrome"} {
```

Comment:
Upspin isn't running on startup for me. Tested on three devices now, I'm not sure what the problem is.

Reply:
Scratch that, it was a permissions issue with my build machine.